ambari-server/src/main/java/org/apache/ambari/server/actionmanager/ActionDBAccessorImpl.java

```diff
@@ -153,6 +155,8 @@
   private Cache<Long, HostRoleCommand> hostRoleCommandCache;
   private long cacheLimit; //may be exceeded to store tasks from one request
 
+  private final ReadWriteLock hrcOperationsLock = new ReentrantReadWriteLock();
```

Comment:
It would be great to add some comment here describing the cases when this lock should be used. The variable name does not say too much

Reply:
Added